### PR TITLE
handle partial fill for bebop

### PR DESCRIPTION
**Pull request by ngocthanh1389**
What is the problem?
Bebop integration support fill partial order, current code only parse original order. So, our trade data was not correct.

What did you do?
Update logic for partial order:
-filledTakerAmount: this is actual amount, taker want to swap
-takerQuoteInfo: old order contain old maker amount

Refer contract: https://vscode.blockscan.com/ethereum/0xbbbbbbb520d69a9775e85b458c58c648259fad5f

### CODE REVIEW

File: pkg/parser/bebop/parser.go [hunk 1]
```diff
@@ -173,54 +195,30 @@ func (p *Parser) ParseFromInternalCall(order storage.TradeLog, internalCall type
...
-
+	var filledTakerAmount *big.Int
+	for _, param := range contractCall.Params {
+		if param.Name != "filledTakerAmount" {
+			continue
+		}
+		v, ok := param.Value.(*big.Int)
+		if ok {
+			filledTakerAmount = v
+		}
```
Comment (linhnt3400) [inline pkg/parser/bebop/parser.go:206]: we should break the for loop after getting the value successfully

File: pkg/parser/bebop/parser.go [hunk 2]
```diff
@@ -283,3 +281,153 @@ func (p *Parser) LogFromExchange(log ethereumTypes.Log) bool {
...
+	}
+
+	for _, param := range contractCall.Params {
+		if param.Name != "takerQuoteInfo" {
+			continue
+		}
+		var oldOrder OldSingleQuote
+		if err := unpackOrder(param.Value, &oldOrder); err == nil {
+			rfqOrder.MakerAmount = oldOrder.MakerAmount
+		}
```
Comment (linhnt3400) [inline pkg/parser/bebop/parser.go:301]: similar to above comment

File: pkg/parser/bebop/parser.go [hunk 3]
```diff
@@ -283,3 +281,153 @@ func (p *Parser) LogFromExchange(log ethereumTypes.Log) bool {
...
+	}
+
+	for _, param := range contractCall.Params {
+		if param.Name != "takerQuoteInfo" {
+			continue
+		}
+		var oldOrder OldMultiQuote
+		if err := unpackOrder(param.Value, &oldOrder); err == nil {
+			rfqOrder.MakerAmounts = oldOrder.MakerAmounts
+		}
```
Comment (linhnt3400) [inline pkg/parser/bebop/parser.go:339]: similar to above comment

File: pkg/parser/bebop/parser.go [hunk 4]
```diff
@@ -283,3 +281,153 @@ func (p *Parser) LogFromExchange(log ethereumTypes.Log) bool {
...
+	}
+
+	for _, param := range contractCall.Params {
+		if param.Name != "takerQuoteInfo" {
+			continue
+		}
+		var oldOrder OldAggregateQuote
+		if err := unpackOrder(param.Value, &oldOrder); err == nil {
+			rfqOrder.MakerAmounts = oldOrder.MakerAmounts
+		}
```
Comment (linhnt3400) [inline pkg/parser/bebop/parser.go:386]: similar to above comment

File: pkg/parser/bebop/parser.go (hunk 1 above)
Comment (ngocthanh1389) [reply pkg/parser/bebop/parser.go:206]: fixed https://github.com/KyberNetwork/tradelogs/pull/64/commits/f51d9861215bf7def1122e25924d420e36ae1d67

File: pkg/parser/bebop/parser.go (hunk 2 above)
Comment (ngocthanh1389) [reply pkg/parser/bebop/parser.go:302]: fixed https://github.com/KyberNetwork/tradelogs/pull/64/commits/f51d9861215bf7def1122e25924d420e36ae1d67

File: pkg/parser/bebop/parser.go (hunk 3 above)
Comment (ngocthanh1389) [reply pkg/parser/bebop/parser.go:341]: fixed https://github.com/KyberNetwork/tradelogs/pull/64/commits/f51d9861215bf7def1122e25924d420e36ae1d67

File: pkg/parser/bebop/parser.go (hunk 4 above)
Comment (ngocthanh1389) [reply pkg/parser/bebop/parser.go:389]: fixed https://github.com/KyberNetwork/tradelogs/pull/64/commits/f51d9861215bf7def1122e25924d420e36ae1d67

**Review by linhnt3400 — APPROVED** [review]

**Pull request merged**